Makefile

```diff
@@ -81,12 +98,32 @@ distclean:
 	-rm -rf deps/icu
 	-rm -rf deps/icu4c*.tgz deps/icu4c*.zip deps/icu-tmp
 	-rm -f $(BINARYTAR).* $(TARBALL).*
+	-rm -rf deps/v8/testing/gmock
+	-rm -rf deps/v8/testing/gtest
 
 check: test
 
 cctest: all
 	@out/$(BUILDTYPE)/$@
 
+v8:
+	tools/make-v8.sh v8
+ifneq (,$(filter $(DESTCPU),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifneq (,$(filter $(ARCH),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifeq ($(ARCH)$(DESTCPU),)
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
```

Comment:
Re: ARCH, just looked at the code in v8 again - turns out that node and v8 use different identifiers for cpu architectures that are not compatible (e.g. node 'x86' vs v8 'ia_32'). I'm thinking about using a different variable $V8_ARCH in order to handle this, since DESTCPU and ARCH are pre-assigned by the makefile using node cpu identifier names.